sdk/eventhub/Azure.Messaging.EventHubs/src/Processor/CheckpointManager.cs

```diff
@@ -0,0 +1,85 @@
+    ///   via <see cref="IPartitionManager" />.
+    /// </summary>
+    ///
+    public class CheckpointManager
+    {
+        /// <summary>
+        ///   Contains information about the partition this instance is associated with.
+        /// </summary>
+        ///
+        private PartitionContext Context { get; }
+
+        /// <summary>
+        ///   Interacts with the storage system, dealing with the creation of checkpoints.
+        /// </summary>
+        ///
+        private IPartitionManager PartitionManager { get; }
+
+        /// <summary>
+        ///   The identifier of the associated <see cref="EventProcessor" /> instance.
+        /// </summary>
+        ///
+        public string OwnerIdentifier { get; }
+
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="CheckpointManager"/> class.
+        /// </summary>
+        ///
+        /// <param name="partitionContext">Contains information about the partition this instance will be associated with.</param>
+        /// <param name="partitionManager">Interacts with the storage system, dealing with the creation of checkpoints.</param>
+        /// <param name="ownerIdentifier">The identifier of the associated <see cref="EventProcessor" /> instance.</param>
+        ///
+        internal CheckpointManager(PartitionContext partitionContext,
+                                   IPartitionManager partitionManager,
+                                   string ownerIdentifier)
+        {
+            Context = partitionContext;
+            PartitionManager = partitionManager;
+            OwnerIdentifier = ownerIdentifier;
+        }
+
+        /// <summary>
+        ///   Updates the checkpoint using the given information for the associated partition and consumer group in the chosen storage service.
+        /// </summary>
+        ///
+        /// <param name="eventData">The event containing the information to be stored in the checkpoint.</param>
+        ///
+        /// <returns>A task to be resolved on when the operation has completed.</returns>
+        ///
+        public Task UpdateCheckpointAsync(EventData eventData) => UpdateCheckpointAsync(eventData.Offset, eventData.SequenceNumber);
+
+        /// <summary>
+        ///   Updates the checkpoint using the given information for the associated partition and consumer group in the chosen storage service.
+        /// </summary>
+        ///
+        /// <param name="offset">The offset of the <see cref="EventData" /> the new checkpoint will be associated with.</param>
+        /// <param name="sequenceNumber">The sequence number assigned to the <see cref="EventData" /> the new checkpoint will be associated with.</param>
+        ///
+        /// <returns>A task to be resolved on when the operation has completed.</returns>
+        ///
+        public async Task UpdateCheckpointAsync(long offset,
```

Comment:
nit:  We should validate the parameters, since this is a public-facing method where values are supplied by users.

Reply:
This method simply creates a Checkpoint from the information provided by the user and send it to `PartitionManager`. `Checkpoint` already does parameter validation. Do you think we should have it in both places just in case?

Reply:
Wasn't putting two and two together that the checkpoint enforced.   No need to double it up, but may be nice to have a comment stating that validation is being delegated to the checkpoint.